docs/sources/shared/wal-data-retention.md

```diff
@@ -74,5 +74,32 @@ If the component falls behind more than one third of the data written since the
 last truncate interval, it is possible for the truncate loop to checkpoint data
 before being pushed to the remote_write endpoint.
 
-[WAL block]: {{< relref "../flow/reference/components/prometheus.remote_write.md/#wal-block" >}}
-[metrics config]: {{< relref "../static/configuration/metrics-config.md" >}}
+### WAL corruption
+
+WAL corruption can occur when Grafana Agent unexpectedly stops while the latest WAL segments
+are still being written to disk. For example, the host computer has a general disk failure
+and crashes before you can stop Grafana Agent and other running services. When you restart Grafana
+Agent, the Agent verifies the WAL, removing any corrupt segments it finds. Sometimes, this repair
+is unsuccessful, and you must manually delete the corrupted WAL to continue.
+
+When the WAL becomes corrupted, Grafana Agent writes error messages such as
+`err="failed to find segment for index"` to the log file.
+
+{{% admonition type="caution" %}}
+Deleting a WAL segment or a WAL file permanently deletes the stored WAL data.
+{{% /admonition %}}
+
+To delete the corrupted WAL:
+
+1. [Stop][] Grafana Agent.
+1. Delete the WAL from disk. By default, the WAL is in the `data-agent` directory located in
```

Comment:
They should be careful to not delete the _entire_ directory, especially in Flow mode, since it has other things. 

They should look for the subdirectory called `wal` (which may be nested inside several other folders in Flow mode) and delete that.

Reply:
@rfratto So something like 

`Find and delete the contents of the wal directory.  By default the wal directory is a subdirectory of data-agent located in the Grafana Agent working directory.`

Might need a little tweaking but something along these lines?

Reply:
Yep. There will be one `wal` directory per:

* (Static mode) Metrics instance running 
* (Flow mode) `prometheus.remote_write` component running